lib/ansible/modules/clustering/consul_acl.py

```diff
@@ -151,56 +158,48 @@ def execute(module):
 
 
 def update_acl(module):
-
     rules = module.params.get('rules')
-    state = module.params.get('state')
     token = module.params.get('token')
     token_type = module.params.get('token_type')
     mgmt = module.params.get('mgmt_token')
     name = module.params.get('name')
     consul = get_consul_api(module, mgmt)
     changed = False
 
-    try:
+    rules = decode_rules_as_yml(rules)
+    rules_as_hcl = encode_rules_as_hcl_string(rules) if len(rules) > 0 else None
 
+    try:
         if token:
             existing_rules = load_rules_for_token(module, consul, token)
-            supplied_rules = yml_to_rules(module, rules)
-            changed = not existing_rules == supplied_rules
+            changed = existing_rules != rules
```

Comment:
Hi! Thanks for this... exactly what I needed.
I'm unsure how this is passing tests, as ansible detects a change each time I run. Looks like you're comparing HCL and JSON here.
If I change this line to:
            changed = encode_rules_as_json(existing_rules) != encode_rules_as_json(rules)
it works as expected (changes when I change a rule, otherwise it doesn't).